namesys/dns.go

```diff
@@ -50,24 +55,46 @@ func (r *DNSResolver) resolveOnce(ctx context.Context, name string) (path.Path,
 	if !isd.IsDomain(segments[0]) {
 		return "", errors.New("not a valid domain name")
 	}
-
 	log.Infof("DNSResolver resolving %s", segments[0])
-	txt, err := r.lookupTXT(segments[0])
+
+	prefixes := []string{"", "_dnslink."}
+
+	resChan := make(chan lookupRes)
+
+	for _, prefix := range prefixes {
+		go workDomain(r, prefix + segments[0], resChan)
```

Comment:
i think this leaks goroutines? will want to do a waitgroup sync, and buffer the channel with `len(prefixes)`